workspaces/jobs/jobs.groovy

```diff
@@ -40,45 +41,46 @@ generateProjectJob.with{
         sshAgent("adop-jenkins-master")
     }
     steps {
-        shell('''#!/bin/bash -e
+      if ( enableInternalLdap == "true" ) {
+          shell('''#!/bin/bash -e
+          # Validate Variables
+          pattern=" |'"
+          if [[ "${PROJECT_NAME}" =~ ${pattern} ]]; then
+            echo "PROJECT_NAME contains a space, please replace with an underscore - exiting..."
+            exit 1
+          fi''')
+          shell('''set -e
+          # LDAP
+          ${WORKSPACE}/common/ldap/generate_role.sh -r "admin" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${ADMIN_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
+          ${WORKSPACE}/common/ldap/generate_role.sh -r "developer" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${DEVELOPER_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
+          ${WORKSPACE}/common/ldap/generate_role.sh -r "viewer" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${VIEWER_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
 
-# Validate Variables
-pattern=" |'"
-if [[ "${PROJECT_NAME}" =~ ${pattern} ]]; then
-	echo "PROJECT_NAME contains a space, please replace with an underscore - exiting..."
-	exit 1
-fi''')
-        shell('''set -e
-# LDAP
-${WORKSPACE}/common/ldap/generate_role.sh -r "admin" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${ADMIN_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
-${WORKSPACE}/common/ldap/generate_role.sh -r "developer" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${DEVELOPER_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
-${WORKSPACE}/common/ldap/generate_role.sh -r "viewer" -n "${WORKSPACE_NAME}.${PROJECT_NAME}" -d "${DC}" -g "${OU_GROUPS}" -p "${OU_PEOPLE}" -u "${VIEWER_USERS}" -f "${OUTPUT_FILE}" -w "${WORKSPACE}"
+          set +e
+          ${WORKSPACE}/common/ldap/load_ldif.sh -h ldap -u "${LDAP_ADMIN_USER}" -p "${LDAP_ADMIN_PASSWORD}" -b "${DC}" -f "${OUTPUT_FILE}"
+          set -e
 
-set +e
-${WORKSPACE}/common/ldap/load_ldif.sh -h ldap -u "${LDAP_ADMIN_USER}" -p "${LDAP_ADMIN_PASSWORD}" -b "${DC}" -f "${OUTPUT_FILE}"
-set -e
+          ADMIN_USERS=$(echo ${ADMIN_USERS} | tr ',' ' ')
+          DEVELOPER_USERS=$(echo ${DEVELOPER_USERS} | tr ',' ' ')
+          VIEWER_USERS=$(echo ${VIEWER_USERS} | tr ',' ' ')
 
-ADMIN_USERS=$(echo ${ADMIN_USERS} | tr ',' ' ')
-DEVELOPER_USERS=$(echo ${DEVELOPER_USERS} | tr ',' ' ')
-VIEWER_USERS=$(echo ${VIEWER_USERS} | tr ',' ' ')
-
-# Gerrit
-for user in $ADMIN_USERS $DEVELOPER_USERS $VIEWER_USERS
-do
-        username=$(echo ${user} | cut -d'@' -f1)
-        ${WORKSPACE}/common/gerrit/create_user.sh -g http://gerrit:8080/gerrit -u "${username}" -p "${username}"
-done''')
-        shell('''#!/bin/bash -ex
```

Comment:
I think this shell block is required for gerrit permissions setup on repositories.